dask/utils.py

```diff
@@ -625,3 +630,83 @@ def insert(tup, loc, val):
     L = list(tup)
     L[loc] = val
     return tuple(L)
+
+
+def infer_storage_options(urlpath, storage_options=None):
+    """ Infer storage options from URL path and merge it with existing storage
+    options.
+
+    Parameters
+    ----------
+    path: str or unicode
+        Either local absolute file path or URL (hdfs://namenode:8020/file.csv)
+    storage_options: dict (optional)
+        Its contents will get merged with the infered information from the
+        given path
+
+    Returns
+    -------
+    Storage options dict.
+
+    >>> so = infer_storage_options('/mnt/datasets/test.csv')
+    >>> so.pop('protocol')
+    'file'
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so
+    {}
+    >>> infer_storage_options('./test.csv')['path']
+    './test.csv'
+    >>> infer_storage_options('../test.csv')['path']
+    '../test.csv'
+    >>> so = infer_storage_options(
+    ...          'hdfs://username:pwd@node:123/mnt/datasets/test.csv?q=1',
+    ...          storage_options={'extra': 'value'})
+    >>> so.pop('protocol')
+    'hdfs'
+    >>> so.pop('username')
+    'username'
+    >>> so.pop('password')
+    'pwd'
+    >>> so.pop('host')
+    'node'
+    >>> so.pop('port')
+    123
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so.pop('url_query')
+    'q=1'
+    >>> so.pop('extra')
+    'value'
+    >>> so
+    {}
```

Comment:
@mrocklin I'm not considering myself as a Hadoop specialist, yet I have over two years of experience with HDFS and PySpark. I liked their [Python API](http://spark.apache.org/examples.html) (but structured data manipulation there is a hassle for our use-case), and there are two options there:

1. Use full URL
2. Use only `hdfs://` + path, while namenode location is infered from `/etc/hadoop/.../hdfs.conf` config file.

I just prefer the full URL, but I see the benefits of the second option as well.